[Teh Kok Hoe] iP

tehkokhoe opened the pull request.

tehkokhoe committed:
Added a FastReader

tehkokhoe committed:
Level-1

tehkokhoe committed:
Level-2

tehkokhoe committed:
added new Class Task

tehkokhoe committed:
Level-3

tehkokhoe committed:
Level-4 Added enum and additional functions

tehkokhoe committed:
added a default constructor

tehkokhoe committed:
Added Deadline class

tehkokhoe committed:
added Event class

tehkokhoe committed:
added ToDo class

tehkokhoe committed:
Remove spaces on toString() method

Fix display

tehkokhoe committed:
Update accessor of enum, add error handling, and change display.

Change the enum accessor so that it is accessible.
Add error handling to prevent program from terminating due to illegal user input.
Change display to look neater.

tehkokhoe committed:
Add test.

Add a semi-automated test to check correctness of code and reduce human error.

tehkokhoe committed:
Add DukeException class

Add Exception class to deal with exceptions unique to program.

tehkokhoe committed:
Add delete feature

Add delete feature to remove a task from the list.

tehkokhoe committed:
Add test cases for delete feature

Add test cases for delete feature to ensure correct display an exception handling

tehkokhoe committed:
Add different display method for tasks for record purposes

Add toRecord method for save method to write in a different display from the output on console

tehkokhoe committed:
Revert "Add different display method for tasks for record purposes"

This reverts commit d355645f3ba732b496733fc2e9bac5066d1bbf13.

tehkokhoe committed:
Add different display method for tasks for record purposes

Add toRecord method for save method to write in a different display from the output on console

tehkokhoe committed:
Add save method and fix to coding standard

Add save method so that whenever list is changed, it is updated to a text file.
Adjust code to fit coding standard.

tehkokhoe committed:
Fix code to coding standard

Adjust code to fit coding standard

bakano98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+import java.util.*;
+
 public class Duke {
+    enum Command {
```
Good to see you're using enums for this! Keep it up.

bakano98 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,50 @@
+public class Task {
+    private String task;
+    private boolean done;
```
According to the coding standards, this should be named `isDone`

bakano98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+        System.out.println("    Hello! I'm Duke");
+        System.out.println("    What can I do for you?");
+        System.out.println("    ____________________________________________________________");
+        boolean StillIn = true;
```
Use camelCase to name variables, and booleans should be of the form `isSomething`. You could possibly rename this to `isStillIn`

bakano98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+            System.out.println("    ____________________________________________________________");
+            try {
+                switch (cmd) {
+                    case BYE:
```
`case` should not be indented. It should be on the same line as `switch`

bakano98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("    ____________________________________________________________");
```
Instead of using 4 spaces, you can consider using `\t` instead for tab

bakano98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+import java.util.*;
```
Two things here: 
1. Keep everything in a package
2. Only import things that are needed to be used. Avoid using wildcard imports

bakano98 commented on src/main/java/FastReader.java:
```diff
@@ -0,0 +1,51 @@
+import java.io.BufferedReader;
```
Not too sure if `FastReader` is required here, but good consideration!

bakano98 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+        boolean StillIn = true;
+        FastReader fr = new FastReader();
+        ArrayList<Task> list = new ArrayList<Task>(100);
+        while(StillIn) {
```
I think you need to put spacing between the brackets and each `while` and each `if`. You only do not use spaces when it is a function/method declaration

bakano98 reviewed (commented):
A few coding standard violations, but other than that, nothing too huge. Remember to put everything into a package.
LGTM after fixing these.

tehkokhoe committed:
Allow dates to be recognized as the object 'Date'

Allow dates to be recognized as 'Date' objects to allow more capabilities like sorting around dates

tehkokhoe committed:
Fix coding standards

tehkokhoe committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

tehkokhoe committed:
Adjust how date is displayed and recorded

Adjust how date works due to complement save implementation

tehkokhoe committed:
Make program more OOP like

Seperate operations into different classes that do specific things

tehkokhoe committed:
Fix how date is parsed in event

Fix how date is parsed in event so that it can read the two dates regardless of whitespace between the seperator '-' and the date

tehkokhoe committed:
Package all classes

Package all classes into one package

tehkokhoe committed:
Add exception handler for invalid commands

Add exception handler for invalid commands by telling user command is invalid and then waits to receive next user input

tehkokhoe committed:
Let parseCommand throw IllegalArgumentException

Let parseCommand throw IllegalArgumentException when receive invalid user input

tehkokhoe committed:
Add JUnit tests

Add JUnit tests for invalid commands and check if bye command quits program

tehkokhoe committed:
Add checkstyle tests

Add checkstyle tests to ensure coding standard

tehkokhoe committed:
Add JavaDoc

Add JavaDoc to allow people to use program easily

tehkokhoe committed:
Fix coding standards

Fix coding standards for improved readability

tehkokhoe committed:
Add find function

Add find function so users can easily filter tasks with specific keywords

tehkokhoe committed:
Merge branch 'branch-A-JavaDoc'

tehkokhoe committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/FastReader.java
#	src/main/java/duke/UI.java

tehkokhoe committed:
Merge branch 'branch-Level-9'

tehkokhoe committed:
Add JavaDoc for find methods

Add JavaDoc for find methods for developer clarity

tehkokhoe committed:
Make a GFMD

tehkokhoe committed:
no message

tehkokhoe committed:
Add GUI

Add GUI so it is more accessible

tehkokhoe committed:
Merge branch 'branch-Level-10'

tehkokhoe committed:
Adjust JavaDoc

Adjust JavaDoc for GUI

tehkokhoe committed:
Add assert

Curent code might not be able to detect some bugs

So that major bugs can be detected when code is changed in the future

Add some asserts to ensure that responses should not be empty for most commands

Assert is low cost and is worth the extra security

tehkokhoe committed:
Merge branch 'branch-A-Assertions'

tehkokhoe committed:
Add duplicate detector

Users can add duplicate tasks into the list

Using equals and contain methods allows duplicate checking

Easy to implement and understand

tehkokhoe committed:
Change UI to make more sense

Change the user picture and input text to be always at the bottom
of the dialog box, and constantly refreshes as users shouldn't
have to know what input they have previously typed, and can infer
by just looking at Duke's output

tehkokhoe committed:
Set theme jekyll-theme-architect

tehkokhoe committed:
Add User Guide

Add User Guide for users to use program comprehensively

tehkokhoe committed:
Merge branch 'master' of https://github.com/tehkokhoe/ip

tehkokhoe committed:
Update dateformat links in User Guide

tehkokhoe committed:
Adjusted coding quality issues

tehkokhoe committed:
Add Ui screenshot
